haskell/stack/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM heroku/cedar:14
+MAINTAINER Gabriel Berke-Williams <gabe@thoughtbot.com>
+
+ENV LANG en_US.UTF-8
+# Stack stores binaries in /root/.local/bin
+ENV PATH /root/.local/bin:$PATH
+
+# Heroku assumes we'll put everything in /app/user
+RUN mkdir -p /app/user
+WORKDIR /app/user
+
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 575159689BEFB442 \
+  && echo 'deb http://download.fpcomplete.com/ubuntu trusty main' > \
+    /etc/apt/sources.list.d/fpco.list \
+  && apt-get update \
+  && apt-get install -y stack \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
+
+# Copy over configuration for building the app
+ONBUILD COPY *.cabal .
```

Comment:
These are `ONBUILD` triggers so that we can `docker build` the `thoughtbot/stack` image without an app while still building an app when used as `FROM thoughtbot/stack`.